lib/Sema/TypeCheckDecl.cpp

```diff
@@ -6693,6 +6693,60 @@ class DeclChecker : public DeclVisitor<DeclChecker> {
                                 desiredAccessScope, access);
       }
       TC.checkConformancesInContext(ED, ED);
+
+      // If we just extended a nominal type with a protocol conformance (which
+      // may have just gotten derived), members which are synthesized might need
+      // to be visited for validation (since the extended type might have
+      // otherwise finished being validated).
```

Comment:
This addition is necessary to perform additional passes on types getting synthesized members in extensions after they've already been validated. For instance, the following code:
```swift
struct X {
    let value: Int
}

extension X : Codable {}
```
performs 4 (very simplified) passes:

1. A first pass on `X` to validate access and function signatures
2. A first pass on `extension X` to validate some access and inherited types
3. A second pass on `X` to do conformance checking
4. A second pass on `extension X` to do conformance checking

It's pass #4 here which actually synthesizes `Codable` methods on `X`. The method bodies, however, are lazily synthesized, so if the function decls are never visited, the bodies are never filled in. This leads to compiling code which references methods which are never compiled in, leading to linker errors.

This forces a pass on synthesized decls in case they haven't been visited before.

Reply:
This doesn't feel right, but I can't quite put my finger on what the right answer is. @DougGregor would probably know.

Reply:
@jrose-apple I agree; if there's a better way to do this, I'm more than happy to rip this out.

Reply:
What is different about Codable that necessitates this logic? Other derived conformances don't need this it seems.

Reply:
@slavapestov `Codable` is so far the only protocol for which we synthesize conformance in a non-constrained manner — `Equatable`, `Hashable`, and `RawRepresentable` are currently only synthesized for `enum`s, and at that, they are automatically synthesized even if you don't request conformance. This case never comes up because we don't support arbitrary `Equatable`/`Hashable` derived conformance, so extending a `struct`/`class` doesn't work. (Yet — when `Equatable`/`Hashable` synthesis lands, it will have the same problem.)